cadl-extension/src/main/java/com/azure/autorest/CadlPlugin.java

```diff
@@ -0,0 +1,165 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.autorest;
+
+import com.azure.autorest.extension.base.jsonrpc.Connection;
+import com.azure.autorest.extension.base.model.Message;
+import com.azure.autorest.extension.base.model.codemodel.CodeModel;
+import com.azure.autorest.extension.base.plugin.JavaSettings;
+import com.azure.autorest.model.clientmodel.ClassType;
+import com.azure.autorest.model.clientmodel.Client;
+import com.azure.autorest.model.clientmodel.ClientModel;
+import com.azure.autorest.model.clientmodel.ClientResponse;
+import com.azure.autorest.model.clientmodel.EnumType;
+import com.azure.autorest.model.clientmodel.IType;
+import com.azure.autorest.model.javamodel.JavaPackage;
+import com.azure.autorest.util.ClientModelUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.lang.reflect.Type;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class CadlPlugin extends Javagen {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CadlPlugin.class);
+
+    @Override
+    public JavaPackage writeToTemplates(CodeModel codeModel, Client client, JavaSettings settings) {
+        return super.writeToTemplates(codeModel, client, settings);
+    }
+
+    @Override
+    protected void writeClientModels(Client client, JavaPackage javaPackage, JavaSettings settings) {
+        // Client model
+        client.getModels().stream()
+                .filter(CadlPlugin::isGeneratingModel)
+                .forEach(model -> javaPackage.addModel(model.getPackage(), model.getName(), model));
+
+        // Enum
+        client.getEnums().stream()
+                .filter(CadlPlugin::isGeneratingModel)
+                .forEach(model -> javaPackage.addEnum(model.getPackage(), model.getName(), model));
+
+        // Response
+        client.getResponseModels().stream()
+                .filter(CadlPlugin::isGeneratingModel)
+                .forEach(model -> javaPackage.addClientResponse(model.getPackage(), model.getName(), model));
+    }
+
+    @Override
+    public void writeFile(String fileName, String content, List<Object> sourceMap) {
+        new File(fileName).getParentFile().mkdirs();
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(fileName))) {
+            writer.write(content);
```

Comment:
Since this is writing the file in on go I don't think a BufferedWriter is necessary

Reply:
I will try the change according to suggestions on read/write.